Subject: Spelling fix
Bug: https://rt.cpan.org/Ticket/Display.html?id=85398
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=85398
Author: Nicholas Bamber <nicholas@periapt.co.uk>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-12-07

Index: libgeo-ip-perl/lib/Geo/IP.pm
===================================================================
--- libgeo-ip-perl.orig/lib/Geo/IP.pm	2014-10-11 03:23:25.401729794 +0200
+++ libgeo-ip-perl/lib/Geo/IP.pm	2014-10-11 03:23:25.401729794 +0200
@@ -5911,7 +5911,7 @@
 
 =item my ( $from, $to ) = $gi->range_by_ip('24.24.24.24');
 
-Returns the start and end of the current network block. The method tries to join several continous netblocks.
+Returns the start and end of the current network block. The method tries to join several continuous netblocks.
 
 =item $api = $gi->api or $api = Geo::IP->api
 
