Index: owdb.pm
===================================================================
--- owdb.pm	(revision 2751)
+++ owdb.pm	(working copy)
@@ -142,11 +142,12 @@
 	scalar %args || die "Invalid args passed to owdb_fetch";
 
 	my $owdbh = $args{'OWDBH'};
+
 	my $owhash = $owdbh->{'OWHASH'};
 	my $nrecs = 0;
 	my $session_done = 0;
 
-	if(!$owdbh->{'DOMORE'}){
+	unless ($owdbh->{'DOMORE'}){
 		return undef;
 	}
 
@@ -276,7 +277,7 @@
                         mesh            => 1,		# the mesh name
                         bucket_width    => 1,		# the mesh's bucket width
                         resolution      => 1,		# the desired resolution
-                        flag_point      => 0,		# the flag point in seconds
+                        flag_points      => 0,		# the flag point in seconds
                         loss_threshold  => 0,		# the loss threshold in seconds
                         start_time      => 1,		# a unix timestamp of the start
                         end_time        => 1,		# a unix timestamp of the end
@@ -297,13 +298,13 @@
         my $bucket_width = $parameters->{bucket_width};
         my $resolution = $parameters->{resolution};
 
-        my $flag_point   = $parameters->{flag_point};
+        my $flag_points   = $parameters->{flag_points};
 	my @flag_points = ();
-	if ($flag_point) {
-		if (ref($flag_point) eq "ARRAY") {
-			@flag_points = @$flag_point;
+	if ($flag_points) {
+		if (ref($flag_points) eq "ARRAY") {
+			@flag_points = @$flag_points;
 		} else {
-			push @flag_points, $flag_point;
+			push @flag_points, $flag_points;
 		}
 	}
 
@@ -462,7 +463,7 @@
                 $test_results{ERR} = 0;
                 $test_results{DUPS} = 0;
 		foreach my $flag_point (@flag_points) {
-			$all_results{FLAGGED}->{$flag_point} = 0;
+			$test_results{FLAGGED}->{$flag_point} = 0;
 		}
                 $test_results{JITTER} = 0;
                 $test_results{HISTOGRAM} = ();
@@ -525,7 +526,7 @@
 	                                }
 				}
 
-                                $test_results{HISTOGRAM}->{$key} = $hist_value;
+                                $test_results{HISTOGRAM}->{$key * $bucket_width} = $hist_value;
                         }
 
 			if ($rolling_loss_packets and $test_results{SENT}) {
