--- Makefile.am
+++ Makefile.am
@@ -3,18 +3,18 @@
 
 INCLUDES = -I$(top_srcdir)/include/ -I$(xerces_include) -I$(top_srcdir)/src/lexer/
 LDFLAGS = -L$(xerces_lib) -R$(xerces_lib)
-LDADD = libxqilla.la
+LDADD = libxqilla10.la
 
 bin_PROGRAMS = xqilla
 noinst_PROGRAMS = basicXQillaUsage xqtsRunner
 
-lib_LTLIBRARIES = libxqilla.la
+lib_LTLIBRARIES = libxqilla10.la
 
 # Library version, see http://sourceware.org/autobook/autobook/autobook_91.html
 # -no-undefined for Cygwin (bug #1639669)
-libxqilla_la_LDFLAGS = -version-info 1:1:0 -no-undefined -lxerces-c
+libxqilla10_la_LDFLAGS = -version-info 1:1:0 -no-undefined -lxerces-c
 
-libxqilla_la_SOURCES = \
+libxqilla10_la_SOURCES = \
 src/framework/ReferenceCounted.cpp \
 src/framework/BaseMemoryManager.cpp \
 src/framework/ProxyMemoryManager.cpp \
