--- configure.in	2007-01-09 02:14:37.000000000 +0100
+++ configure.in	2007-11-10 22:27:28.000000000 +0100
@@ -69,18 +69,18 @@
 if test "$with_xerces" = "no"; then
         with_xerces="/usr/local"
 fi
-if test `ls $with_xerces/lib/libxerces-c.*2[[6-7]].* 2>/dev/null | wc -l` -gt 0 ; then
-        AC_MSG_CHECKING([for Xerces C++ version from source tree])
-        xerces_version=`ls "$with_xerces"/lib/libxerces-c.*[[0-9]][[0-9]].* | tail -1 | sed 's/.*xerces-c.*\([[0-9]]\)\([[0-9]]\).*/\1.\2/'`
+if test `ls $libdir/libxerces-c.*2[[6-7]].* 2>/dev/null | wc -l` -gt 0 ; then
+        AC_MSG_CHECKING([for Xerces C++ version from installed library])
+        xerces_version=`ls "$libdir"/libxerces-c.*[[0-9]][[0-9]].* | tail -1 | sed 's/.*xerces-c.*\([[0-9]]\)\([[0-9]]\).*/\1.\2/'`
         AC_MSG_RESULT([$xerces_version])
-        xerces_lib="$with_xerces/lib"
+        xerces_lib="$libdir"
         xerces_include="$with_xerces/src"
         xerces_library="xerces-c"
-elif test `ls $with_xerces/lib/libxerces-c*2[[6-7]].* 2>/dev/null | wc -l` -gt 0 ; then
-        AC_MSG_CHECKING([for Xerces C++ version from source tree])
-        xerces_version=`ls "$with_xerces"/lib/libxerces-c*[[0-9]][[0-9]].* | tail -1 | sed 's/.*xerces-c*\([[0-9]]\)\([[0-9]]\).*/\1.\2/'`
+elif test `ls $libdir/libxerces-c*2[[6-7]].* 2>/dev/null | wc -l` -gt 0 ; then
+        AC_MSG_CHECKING([for Xerces C++ version from installed library])
+        xerces_version=`ls "$libdir"/libxerces-c*[[0-9]][[0-9]].* | tail -1 | sed 's/.*xerces-c*\([[0-9]]\)\([[0-9]]\).*/\1.\2/'`
         AC_MSG_RESULT([$xerces_version])
-        xerces_lib="$with_xerces/lib"
+        xerces_lib="$libdir"
         xerces_include="$with_xerces/src"
         xerces_library="xerces-c"
 else
