Index: firewalld/config/firewalld.service.in
===================================================================
--- firewalld.orig/config/firewalld.service.in	2013-03-28 22:22:08.995826857 +0100
+++ firewalld/config/firewalld.service.in	2013-03-28 22:22:22.056028858 +0100
@@ -6,8 +6,7 @@
 Conflicts=iptables.service ip6tables.service ebtables.service
 
 [Service]
-EnvironmentFile=-/etc/sysconfig/firewalld
-ExecStart=@sbindir@/firewalld --nofork --nopid $FIREWALLD_ARGS
+ExecStart=@sbindir@/firewalld --nofork --nopid
 ExecReload=/bin/kill -HUP $MAINPID
 # supress to log debug and error output also to /var/log/messages
 StandardOutput=null
