Description: Inlude staticfiles in INSTALLED_APPS only once.
Author: Brian May <brian@microcomaustralia.com.au>
Bug-Debian: https://bugs.debian.org/758381
Origin: https://github.com/brianmay/django-guardian/blob/master/debian/patches/0006-Inlude-staticfiles-in-INSTALLED_APPS-only-once.patch
Last-Update: 2014-08-25

--- django-guardian-1.2.4.orig/example_project/settings.py
+++ django-guardian-1.2.4/example_project/settings.py
@@ -40,10 +40,6 @@ INSTALLED_APPS = (
     'core',
     'integration_tests',
 )
-if django.VERSION < (1, 3):
-    INSTALLED_APPS += ('staticfiles',)
-else:
-    INSTALLED_APPS += ('django.contrib.staticfiles',)
 
 if 'GUARDIAN_NO_TESTS_APP' in os.environ:
     _apps = list(INSTALLED_APPS)
