--- a/Makefile.in
+++ b/Makefile.in
@@ -676,7 +676,7 @@
 
 
 sdatadir:
-	test -z "$(ndtdir)/serverdata" || $(mkdir_p) "$(ndtdir)/serverdata"
+	test -z "$(DESTDIR)$(ndtdir)/serverdata" || $(mkdir_p) "$(DESTDIR)$(ndtdir)/serverdata"
 
 install: sdatadir install-recursive
 
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -1656,13 +1656,13 @@
 
 
 install-ln:
-	@if test -f "$(prefix)/sbin/web100srv" && test ! -L "$(prefix)/sbin/ndtd"; then \
-		echo '(cd $(prefix)/sbin && $(LN_S) web100srv ndtd )' ; \
-		(cd $(prefix)/sbin && $(LN_S) web100srv ndtd ); \
+	@if test -f "$(DESTDIR)$(prefix)/sbin/web100srv" && test ! -L "$(DESTDIR)$(prefix)/sbin/ndtd"; then \
+		echo '(cd $(DESTDIR)$(prefix)/sbin && $(LN_S) web100srv ndtd )' ; \
+		(cd $(DESTDIR)$(prefix)/sbin && $(LN_S) web100srv ndtd ); \
 	else :; fi
-	@if test -f "$(prefix)/bin/web100clt" && test ! -L "$(prefix)/bin/ndtclt"; then \
-		echo '(cd $(prefix)/bin && $(LN_S) web100clt ndtclt )' ; \
-		(cd $(prefix)/bin && $(LN_S) web100clt ndtclt ); \
+	@if test -f "$(DESTDIR)$(prefix)/bin/web100clt" && test ! -L "$(DESTDIR)$(prefix)/bin/ndtclt"; then \
+		echo '(cd $(DESTDIR)$(prefix)/bin && $(LN_S) web100clt ndtclt )' ; \
+		(cd $(DESTDIR)$(prefix)/bin && $(LN_S) web100clt ndtclt ); \
 	else :; fi
 
 install: install-am install-ln
