--- dist/aclocal/options.ac	2006-06-13 21:32:29.000000000 +0200
+++ dist/aclocal/options.ac	2007-11-12 20:55:02.000000000 +0100
@@ -55,23 +55,24 @@
 AC_MSG_CHECKING([if --with-berkeleydb=DIR option specified])
 AC_ARG_WITH(berkeleydb,
 	[AC_HELP_STRING([--with-berkeleydb=DIR],
-			[Path of Berkeley DB. [DIR="/usr/local/BerkeleyDB.@DB_VERSION_MAJOR@.@DB_VERSION_MINOR@"]])],
+			[Path of Berkeley DB. [DIR="/usr"]])],
 	[with_berkeleydb="$withval"], [with_berkeleydb="no"])
 AC_MSG_RESULT($with_berkeleydb)
 # If --with-berkeleydb isn't specified, assume it's here | wc -l` -gt 0
 if test "$with_berkeleydb" = "no"; then
-	with_berkeleydb="/usr/local/BerkeleyDB.@DB_VERSION_MAJOR@.@DB_VERSION_MINOR@"
+	with_berkeleydb="/usr"
 fi
 with_berkeleydb=`cd $with_berkeleydb && pwd`
-if test `ls "$with_berkeleydb"/lib/libdb_cxx-*.la 2>/dev/null | wc -l` -gt 0 ; then
-	AC_MSG_CHECKING([for Berkeley DB version from install tree])
-	db_version=`ls "$with_berkeleydb"/lib/libdb_cxx-*.la | sed 's/.*db_cxx-\(.*\).la/\1/'`
+if test `ls "$libdir"/libdb_cxx-?.?.so 2>/dev/null | wc -l` -gt 0 ; then
+	AC_MSG_CHECKING([for Berkeley DB version from installed library])
+	db_version=`ls "$libdir"/libdb_cxx-?.?.so | sed 's/.*db_cxx-\(.*\).so/\1/'`
 	AC_MSG_RESULT([$db_version])
-	if test "$db_version" != "4.3" && test "$db_version" != "4.4" && test "$db_version" != "4.5"; then
+	if test "$db_version" != "4.3" && test "$db_version" != "4.4" && test "$db_version" != "4.5" && test "$db_version" != "4.6"; then
 		AC_MSG_ERROR([$with_berkeleydb is version $db_version, which is not a supported version of Berkeley DB. The version must be at least 4.3])
 	fi
 	echo "$CPPFLAGS" | grep "$with_berkeleydb/include" >/dev/null 2>&1 || CPPFLAGS="$CPPFLAGS -I$with_berkeleydb/include"
-	LIBS="$LIBS $with_berkeleydb/lib/libdb_cxx-$db_version.la"
+	LIBS="$LIBS -L$libdir -ldb_cxx-$db_version"
+	LIBXSO_LIBS="$LIBXSO_LIBS -L$libdir -ldb_cxx-$db_version"
 
 	# find test lib locations, if specified, and possible
 	# it could be safer to require that --with-berkeleydb point to a build
@@ -100,7 +100,7 @@
 	    fi
 	fi
 
-	ADDITIONAL_CLASSPATH="$ADDITIONAL_CLASSPATH:$with_berkeleydb/lib/db.jar"
+	ADDITIONAL_CLASSPATH="$ADDITIONAL_CLASSPATH:/usr/share/java/db.jar"
 elif test `ls "$with_berkeleydb"/build_unix/libdb_cxx-*.la 2>/dev/null | wc -l` -gt 0 ; then
 	AC_MSG_CHECKING([for Berkeley DB version from build tree])
 	db_version=`ls "$with_berkeleydb"/build_unix/libdb_cxx-*.la | sed 's/.*db_cxx-\(.*\).la/\1/'`
@@ -146,10 +146,11 @@
 	with_xqilla="@XQILLA_HOME@"
 fi
 with_xqilla=`cd $with_xqilla && pwd`
-# Try as an install directory
-if test `ls "$with_xqilla"/lib/libxqilla*.la 2>/dev/null | wc -l` -gt 0 ; then
-	echo "$CPPFLAGS" | grep "$with_xqilla/include" >/dev/null 2>&1 || CPPFLAGS="$CPPFLAGS -I$with_xqilla/include"
-	LIBS="$LIBS -L$with_xqilla/lib -lxqilla"
+# Try from installed library
+if test `ls "$libdir"/libxqilla10.so.1.0.? 2>/dev/null | wc -l` -gt 0 ; then
+	echo "$CPPFLAGS" | grep "$with_xqilla/include" >/dev/null 2>&1 || CPPFLAGS="$CPPFLAGS -I$with_xqilla/include/xqilla10"
+	LIBS="$LIBS -L$libdir -lxqilla10"
+	LIBXSO_LIBS="$LIBXSO_LIBS -L$libdir -lxqilla10"
 # Try as a source directory, built in the top level directory
 elif test `ls "$with_xqilla"/libxqilla*.la 2>/dev/null | wc -l` -gt 0 ; then
   # If we have been given a build directory inside the XQilla top
@@ -177,12 +177,13 @@
 fi
 
 # specifically test for 2.6 or 2.7.  This needs to change as releases change
-if test `ls $with_xerces/lib/libxerces-c.*2[[6-7]].* 2>/dev/null | wc -l` -gt 0 ; then
-	AC_MSG_CHECKING([for Xerces C++ version from install tree])
-	xerces_version=`ls "$with_xerces"/lib/libxerces-c.*[[0-9]][[0-9]].* | tail -1 | sed 's/.*xerces-c.*\([[0-9]]\)\([[0-9]]\).*/\1.\2/'`
+if test `ls $libdir/libxerces-c.*2[[6-7]].* 2>/dev/null | wc -l` -gt 0 ; then
+	AC_MSG_CHECKING([for Xerces C++ version from installed library])
+	xerces_version=`ls "$libdir"/libxerces-c.*[[0-9]][[0-9]].* | tail -1 | sed 's/.*xerces-c.*\([[0-9]]\)\([[0-9]]\).*/\1.\2/'`
 	with_xerces=`cd $with_xerces && pwd`
 	echo "$CPPFLAGS" | grep "$with_xerces/include" >/dev/null 2>&1 || CPPFLAGS="$CPPFLAGS -I$with_xerces/include"
-	LIBS="$LIBS -L$with_xerces/lib -R$with_xerces/lib -lxerces-c"
+	LIBS="$LIBS -L$libdir -lxerces-c"
+	LIBXSO_LIBS="$LIBXSO_LIBS -L$libdir -lxerces-c"
 	AC_MSG_RESULT([$xerces_version])
 elif test `ls $with_xerces/lib/libxerces-c*2[[6-7]].* 2>/dev/null | wc -l` -gt 0 ; then
 	AC_MSG_CHECKING([for Xerces C++ version from install tree])
--- dist/Makefile.in	2007-01-19 16:52:44.000000000 +0100
+++ dist/Makefile.in	2007-11-12 20:54:36.000000000 +0100
@@ -327,8 +327,8 @@
 uninstall: uninstall_include uninstall_lib uninstall_utilities uninstall_docs
 
 install_setup:
-	@test -d $(prefix) || \
-	    ($(mkdir) -p $(prefix) && $(chmod) $(dmode) $(prefix))
+	@test -d $(DESTDIR)$(prefix) || \
+	    ($(mkdir) -p $(DESTDIR)$(prefix) && $(chmod) $(dmode) $(DESTDIR)$(prefix))
 
 INCLUDE_NAMES= DbXml.hpp DbXmlFwd.hpp XmlQueryExpression.hpp XmlContainer.hpp \
 	XmlDocument.hpp XmlException.hpp XmlIndexSpecification.hpp \
@@ -342,22 +342,22 @@
 INCLUDE_FILES= $(patsubst %,$(srcdir)/include/dbxml/,$(INCLUDE_NAMES))
 
 install_include:
-	@echo "Installing DB XML include files: $(includedir) ..."
-	@test -d $(includedir) || \
-	    ($(mkdir) -p $(includedir) && $(chmod) $(dmode) $(includedir))
-	@cd $(srcdir)/include/dbxml/ && $(cp) -pf $(INCLUDE_NAMES) $(includedir)
-	@cd $(includedir) && $(chmod) $(fmode) $(INCLUDE_NAMES)
+	@echo "Installing DB XML include files: $(DESTDIR)$(includedir) ..."
+	@test -d $(DESTDIR)$(includedir) || \
+	    ($(mkdir) -p $(DESTDIR)$(includedir) && $(chmod) $(dmode) $(DESTDIR)$(includedir))
+	@cd $(srcdir)/include/dbxml/ && $(cp) -pf $(INCLUDE_NAMES) $(DESTDIR)$(includedir)
+	@cd $(DESTDIR)$(includedir) && $(chmod) $(fmode) $(INCLUDE_NAMES)
 
 uninstall_include:
-	@cd $(includedir) && $(rm) -f $(INCLUDE_NAMES)
+	@cd $(DESTDIR)$(includedir) && $(rm) -f $(INCLUDE_NAMES)
 
 install_lib:
-	@echo "Installing DB XML library: $(libdir) ..."
-	@test -d $(libdir) || \
-	    ($(mkdir) -p $(libdir) && $(chmod) $(dmode) $(libdir))
-	@cd $(libdir) && $(rm) -f $(LIB_INSTALL_FILE_LIST)
-	@$(INSTALLER) $(libxso_target) $(install_tcl) $(install_java) $(libdir)
-	@(cd $(libdir) && \
+	@echo "Installing DB XML library: $(DESTDIR)$(libdir) ..."
+	@test -d $(DESTDIR)$(libdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(libdir) && $(chmod) $(dmode) $(DESTDIR)$(libdir))
+	@cd $(DESTDIR)$(libdir) && $(rm) -f $(LIB_INSTALL_FILE_LIST)
+	@$(INSTALLER) $(libxso_target) $(install_tcl) $(install_java) $(DESTDIR)$(libdir)
+	@(cd $(DESTDIR)$(libdir) && \
 	    test -f $(libxso) && $(ln) -s $(libxso) $(libxso_default); \
 	    test -f $(libxso) && $(ln) -s $(libxso) $(libxso_major); \
 	    test -f $(libxso_static) && $(ln) -s $(libxso_static) $(libcxx); \
@@ -368,26 +368,26 @@
 	    test -f $(libjso) && $(ln) -s $(libjso) $(libjso_g); \
 	    $(LIBJSO_LN_INSTALL)) || true
 	@(test -f $(libj_jarfile) && \
-	    $(cp) $(libj_jarfile) $(libdir) && \
-	    $(chmod) $(fmode) $(libdir)/$(libj_jarfile)) || true
+	    $(cp) $(libj_jarfile) $(DESTDIR)$(libdir) && \
+	    $(chmod) $(fmode) $(DESTDIR)$(libdir)/$(libj_jarfile)) || true
 
 uninstall_lib:
-	@cd $(libdir) && $(rm) -f $(LIB_INSTALL_FILE_LIST)
+	@cd $(DESTDIR)$(libdir) && $(rm) -f $(LIB_INSTALL_FILE_LIST)
 
 install_utilities:
-	echo "Installing DB XML utilities: $(bindir) ..."
-	@test -d $(bindir) || \
-	    ($(mkdir) -p $(bindir) && $(chmod) $(dmode) $(bindir))
+	echo "Installing DB XML utilities: $(DESTDIR)$(bindir) ..."
+	@test -d $(DESTDIR)$(bindir) || \
+	    ($(mkdir) -p $(DESTDIR)$(bindir) && $(chmod) $(dmode) $(DESTDIR)$(bindir))
 	@for i in $(UTIL_PROGS); do \
-		$(rm) -f $(bindir)/$$i $(bindir)/$$i.exe; \
+		$(rm) -f $(DESTDIR)$(bindir)/$$i $(DESTDIR)$(bindir)/$$i.exe; \
 		test -f $$i.exe && i=$$i.exe || true; \
-		$(INSTALLER) $$i $(bindir)/$$i; \
-		test -f $(strip) && $(strip) $(bindir)/$$i || true; \
-		$(chmod) $(emode) $(bindir)/$$i; \
+		$(INSTALLER) $$i $(DESTDIR)$(bindir)/$$i; \
+		test -f $(strip) && $(strip) $(DESTDIR)$(bindir)/$$i || true; \
+		$(chmod) $(emode) $(DESTDIR)$(bindir)/$$i; \
 	done
 
 uninstall_utilities:
-	@(cd $(bindir); for i in $(UTIL_PROGS); do \
+	@(cd $(DESTDIR)$(bindir); for i in $(UTIL_PROGS); do \
 		$(rm) -f $$i $$i.exe; \
 	done)
 
@@ -395,14 +395,14 @@
 	api_c api_cxx api_tcl collections gsg gsg_db_rep gsg_txn gsg_xml_txn \
 	gsg_xml images index.html license java ref ref_xml intro_xml utility
 install_docs:
-	@echo "Installing documentation: $(docdir) ..."
-	@test -d $(docdir) || \
-	    ($(mkdir) -p $(docdir) && $(chmod) $(dmode) $(docdir))
-	@cd $(docdir) && $(rm) -rf $(DOCLIST)
-	@cd $(srcdir)/docs && $(cp) -pr $(DOCLIST) $(docdir)/
+	@echo "Installing documentation: $(DESTDIR)$(docdir) ..."
+	@test -d $(DESTDIR)$(docdir) || \
+	    ($(mkdir) -p $(DESTDIR)$(docdir) && $(chmod) $(dmode) $(DESTDIR)$(docdir))
+	@cd $(DESTDIR)$(docdir) && $(rm) -rf $(DOCLIST)
+	@cd $(srcdir)/docs && $(cp) -pr $(DOCLIST) $(DESTDIR)$(docdir)/
 
 uninstall_docs:
-	@cd $(docdir) && $(rm) -rf $(DOCLIST)
+	@cd $(DESTDIR)$(docdir) && $(rm) -rf $(DOCLIST)
 
 ##################################################
 # Remaining standard Makefile targets.
