--- src/python/dbxml_python_wrap.cpp	2006-12-15 23:38:53.000000000 +0100
+++ src/python/dbxml_python_wrap.cpp	2007-11-12 21:00:35.000000000 +0100
@@ -2591,6 +2591,11 @@
 #include <errno.h>
 #include <fstream>
 
+#ifndef DB_READ_UNCOMMITTED
+#define DB_READ_UNCOMMITTED DB_DIRTY_READ
+#define DB_READ_COMMITTED DB_DEGREE_2
+#endif
+
 using namespace DbXml;
 
 class XmlIndexDeclaration {
--- src/python/setup.py.template	2006-12-15 23:39:10.000000000 +0100
+++ src/python/setup.py.template	2007-11-12 21:03:45.000000000 +0100
@@ -16,22 +16,14 @@
 # Windows: may require further editing to reflect site specifics.
 #
 if os.name == "posix":
-  db_home = "@DB_DIR@"
-  xerces_home = "@XERCES_DIR@"
   xqilla_home = "@XQILLA_DIR@"
   LFLAGS = os.environ.get('LFLAGS', [])
 else:
-  xerces_home = "../@XERCES_WINHOME@"
   xqilla_home = "../@XQILLA_HOME@"
-  db_home = "../@DB_HOME@"

 for arg in sys.argv:
-  if arg.startswith('--with-berkeleydb='):
-    db_home = arg.split('=')[1]
-  elif arg.startswith('--with-xerces='):
-    xerces_home = arg.split('=')[1]
-  elif arg.startswith('--with-xqilla='):
-    xqillahome = arg.split('=')[1]
+  if arg.startswith('--with-xqilla='):
+    xqilla_home = arg.split('=')[1]
   elif arg.startswith('--lflags='):
     LFLAGS = arg.split('=')[1].split()
     sys.argv.remove(arg)
@@ -48,16 +40,12 @@

 if os.name == "posix":
   INCLUDES =    ["../../include",
-                 os.path.join(db_home, "include")]
+                 os.path.join(xqilla_home, "include")]

-  LIBDIRS =     ["../../build_unix/.libs",
-                 os.path.join(db_home, "lib"),
-                 os.path.join(xqilla_home, "lib"),
-                 os.path.join(xerces_home, "lib")]
+  LIBDIRS =     ["../../build_unix/.libs"]

-  LIBS =        ["dbxml-@DBXML_VERSION_MAJOR@",
-                 "db_cxx-@DB_VERSION_MAJOR@",
-                 "xqilla", "xerces-c"]
+  LIBS =        ["dbxml-@DBXML_VERSION_MAJOR@.@DBXML_VERSION_MINOR@",
+                 "db_cxx", "xqilla10", "xerces-c"]

   DATAFILES =   []

@@ -80,7 +68,6 @@
               ext_modules = [Extension("_dbxml", ["dbxml_python_wrap.cpp"],
                                        include_dirs = INCLUDES,
                                        library_dirs = LIBDIRS,
-                                       runtime_library_dirs = LIBDIRS,
                                        libraries = LIBS,
                                        extra_link_args = lflags_arg
                                        )],
