--- Makefile.PL.orig	2023-05-09 15:40:54.127649239 -0400
+++ Makefile.PL	2023-05-09 15:42:16.724486651 -0400
@@ -1,8 +1,6 @@
 use 5.008003;
 use ExtUtils::MakeMaker;
 
-use Canary::Stability JSON::XS => 1, 5.008003;
-
 WriteMakefile(
     dist	 => {
 	             PREOP	=> 'pod2text XS.pm | tee README >$(DISTVNAME)/README; chmod -R u=rwX,go=rX . ;',
@@ -16,6 +14,6 @@
        common::sense     => 0,
        Types::Serialiser => 0,
     },
-    CONFIGURE_REQUIRES => { ExtUtils::MakeMaker => 6.52, Canary::Stability => 0 },
+    CONFIGURE_REQUIRES => { ExtUtils::MakeMaker => 6.52 },
 );
 
