--- buildall.sh	2009-03-31 23:35:43.000000000 -0400
+++ buildall.sh.new	2009-03-31 23:40:04.000000000 -0400
@@ -474,12 +474,13 @@
     echo "for configuration and build instructions"
     cd $DBXML_DIR/src/perl
     if [ $DO_CONFIGURE = "yes" ]; then
-	$PERL Makefile.PL
+#	$PERL Makefile.PL
+	$PERL Makefile.PL PREFIX=$DBXML_INSTALL_DIR
     fi
     echo Building Perl
     $MAKE || exit $?
-    echo Testing Perl
-    $MAKE test || exit $?
+#    echo Testing Perl
+#    $MAKE test || exit $?
     if [ $DO_INSTALL = "yes" ]; then
 	echo Installing Perl
 	$MAKE install || exit $?
