--- a/t/testmodule.t
+++ b/t/testmodule.t
@@ -36,6 +36,7 @@
 
     $client = new IO::Socket::SSL(PeerAddr => $SSL_SERVER_ADDR,
 				  PeerPort => $SSL_SERVER_PORT,
+				  SSL_version => 'TLSv1',
 				  SSL_verify_mode => 0x01,
 				  SSL_ca_file => "certs/test-ca.pem");
 
@@ -58,7 +59,7 @@
 				   Timeout => 30,
 				   ReuseAddr => 1,
 				   SSL_verify_mode => 0x00,
-				   SSL_ca_file => "certs/test-ca.pem",
+				   SSL_key_file => "certs/server-key.pem",
 				   SSL_cert_file => "certs/server-cert.pem");
 
 if (!$server) {
